### PR TITLE
Bump 1.26.2

### PR DESCRIPTION
**Pull request by dinhxuanvu**
Merge tag 'v0.26.2' into bump-1.26.2
Kubernetes release v1.26.2

Based on https://github.com/kubernetes/kubernetes/releases/tag/v1.26.2

Signed-off-by: Vu Dinh <vudinh@outlook.com>

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
@sanchezl PTAL

**Comment by dinhxuanvu** [comment]
One conflict which I resolved was: https://github.com/openshift/kubernetes-apiserver/pull/41/files#diff-b2632c93071b4e924f1bd32172b02de47fb97ad9f1924a7b8c3f1f3c70af36cbL181

File: pkg/endpoints/handlers/delete.go [hunk 1]
```diff
@@ -178,11 +178,9 @@ func DeleteCollection(r rest.CollectionDeleter, checkBody bool, scope *RequestSc
 			return
 		}
 
-		// enforce a timeout of at most requestTimeoutUpperBound (34s) or less if the user-provided
-		// timeout inside the parent context is lower than requestTimeoutUpperBound.
-		ctx, cancel := filters.RequestContextWithUpperBoundOrWorkAroundOurBrokenCaseWhereTimeoutWasNotAppliedYet(req, requestTimeoutUpperBound)
-		defer cancel()
-
```
Comment (sanchezl) [inline pkg/endpoints/handlers/delete.go:185]: @tkashem Do we still need to worry about the user specified timeout in the request?

Comment (tkashem) [reply pkg/endpoints/handlers/delete.go:185]: yes, we still do, we did not remove all, i think we still have this timeout enforcement on delete (an item), and create. I have plans to fix it in 1.28

File: pkg/endpoints/handlers/delete_test.go [hunk 2]
```diff
@@ -204,3 +208,73 @@ func TestDeleteCollection(t *testing.T) {
...
+	if _, hasDeadline := request.Context().Deadline(); hasDeadline {
+		t.Fatalf("expected request context to not have any deadline")
+	}
+
+	recorder := httptest.NewRecorder()
+	handler.ServeHTTP(recorder, request)
+	if atomic.LoadInt32(&invokedGot) != 1 {
+		t.Errorf("expected collection deleter to be invoked")
+	}
+	// NOTE(dinhxuanvu): The deadline context is kept as a carry-patch.
```
Comment (dinhxuanvu) [inline pkg/endpoints/handlers/delete_test.go:248]: @sanchezl @tkashem Given we keep the deadline context as carry-patch, I have to comment out a section of a new test case to ensure it pass. FYI.

**Comment by dinhxuanvu** [comment]
@sanchezl PTAL

**Review by sanchezl — APPROVED** [review]
Please create a proof PR.

**Comment by dinhxuanvu** [comment]
Proof downstream PR: https://github.com/openshift/openshift-apiserver/pull/360

**Review by sanchezl — APPROVED** [review]
/lgtm

**Pull request merged**